ddsmt.py

```diff
@@ -128,13 +128,33 @@ def _test ():
     return exitcode == g_golden_exit
 
 
-def _filter_scopes (filter_fun, root = None):
+def _filter_scopes (filter_fun, root = None, bfs = False):
+    """_filter_scopes(filter_fun, root, bfs)
+   
+       Collect a list of scope nodes that fit a condition defined by given filtering 
+       function filter_fun. Nodes are added to the list in the order that they 
+       are visited in a depth-first search descending from a given root.
+
+       If no root is specified, search will begin from the root of the entire formula.
+
+       If g_args.bfs or bfs are True, nodes are visited in a breadth-first search
+       instead.
+
+       :filter_fun: Boolean function that returns True if a node should be added.
+       :roots: List of nodes from which to begin searching.
+       :bfs: Bool indicating whether to use breadth-first search.
+       :return: List of scope nodes that fit the filtering condition.
+
+    """
     global g_smtformula
     assert (g_smtformula)
     scopes = []
     to_visit = [root if root else g_smtformula.scopes]
     while to_visit:
-        cur = to_visit.pop()
+        if bfs or g_args.bfs:
```

Comment:
Can't we pass `g_args.bfs` to _filter_scopes (as option `bfs`) instead of checking `g_args.bfs` here?